lib/jsonwp-proxy/proxy.js

```diff
@@ -19,15 +20,34 @@ class JWProxy {
       port: 4444,
       base: '/wd/hub',
       sessionId: null,
-      timeout: REQUEST_TIMEOUT,
+      timeout: DEFAULT_REQUEST_TIMEOUT,
     }, opts);
     this.scheme = this.scheme.toLowerCase();
+    this._activeRequests = [];
   }
 
   // abstract the call behind a member function
   // so that we can mock it in tests
   async request (...args) {
-    return await request(...args);
+    const currentRequest = request(...args);
+    this._activeRequests.push(r);
+    return await currentRequest.finally(() => {
+                    _.remove(this._activeRequests, (r) => {
+                      r === currentRequest;
+                    });
+                  });
+  }
+
+  hasActiveRequests () {
+    return this._activeRequests.length > 0;
+  }
+
+  async cancelActiveRequests () {
+    await B.Promise.map(this._activeRequests, (r) => {
+      r.cancel();
```

Comment:
Because with a synchronous call there is no chance for there to be handoff of control. So the first call to the inner method will finish, then the next will be called, then the next, sequentially. There is nothing promise-ey going on in there.